`postrun` hooks of uninstalled plugins are executed

**Describe the bug**

When uninstalling a plugin that has a `postrun` hook, `@oclif/core` still attempts to run the hook even though that code path no longer exists.

```
~ » sf plugins uninstall sfdx-hardis                                                                                                                                                                                                                
Uninstalling sfdx-hardis... ⢿ [2/2] Regenerating lockfile and installing missing dependencies...
Uninstalling sfdx-hardis... done
 ›   ModuleLoadError: [MODULE_NOT_FOUND] require failed to load /Users/mdonnalley/.local/share/sf/node_modules/sfdx-hardis/lib/hooks/postrun/notify: Cannot find module '/Users/mdonnalley/.local/share/sf/node_modules/sfdx-hardis/lib/hooks/postrun/notify'
 ›   Require stack:
 ›   - /Users/mdonnalley/.local/share/sf/client/2.10.2-1c35561/node_modules/@oclif/core/lib/module-loader.js
 ›   - /Users/mdonnalley/.local/share/sf/client/2.10.2-1c35561/node_modules/@oclif/core/lib/help/util.js
 ›   - /Users/mdonnalley/.local/share/sf/client/2.10.2-1c35561/node_modules/@oclif/core/lib/help/formatter.js
 ›   - /Users/mdonnalley/.local/share/sf/client/2.10.2-1c35561/node_modules/@oclif/core/lib/help/command.js
 ›   - /Users/mdonnalley/.local/share/sf/client/2.10.2-1c35561/node_modules/@oclif/core/lib/help/index.js
 ›   - /Users/mdonnalley/.local/share/sf/client/2.10.2-1c35561/node_modules/@oclif/core/lib/flags.js
 ›   - /Users/mdonnalley/.local/share/sf/client/2.10.2-1c35561/node_modules/@oclif/core/lib/cli-ux/styled/table.js
 ›   - /Users/mdonnalley/.local/share/sf/client/2.10.2-1c35561/node_modules/@oclif/core/lib/cli-ux/styled/index.js
 ›   - /Users/mdonnalley/.local/share/sf/client/2.10.2-1c35561/node_modules/@oclif/core/lib/cli-ux/index.js
 ›   - /Users/mdonnalley/.local/share/sf/client/2.10.2-1c35561/node_modules/@oclif/core/lib/command.js
 ›   - /Users/mdonnalley/.local/share/sf/client/2.10.2-1c35561/node_modules/@oclif/core/lib/index.js
 ›   - /Users/mdonnalley/.local/share/sf/client/2.10.2-1c35561/dist/cli.js
 ›   - /Users/mdonnalley/.local/share/sf/client/2.10.2-1c35561/bin/run
 ›   Code: MODULE_NOT_FOUND
```

**To Reproduce**
1. `sf plugins install sfdx-hardis`
2. `sf plugins uninstall sfdx-hardis`

**Expected behavior**
The `postrun` hook of the uninstalled plugin should not be run.

--- a/src/config/config.ts
+++ b/src/config/config.ts
@@ -604,6 +604,13 @@ export class Config implements IConfig {
     await this.runHook('prerun', {Command: command, argv})
 
     const result = (await command.run(argv, this)) as T
+    // If plugins:uninstall was run, we need to remove all the uninstalled plugins
+    // from this.plugins so that the postrun hook doesn't attempt to run any
+    // hooks that might have existed in the uninstalled plugins.
+    if (c.id === 'plugins:uninstall') {
+      for (const arg of argv) this.plugins.delete(arg)
+    }
+
     await this.runHook('postrun', {Command: command, argv, result})
 
     marker?.addDetails({command: id, plugin: c.pluginName!})

